### PR TITLE
GCC 10 and settable flags

### PR DESCRIPTION
**Pull request by kgerheiser**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -8,15 +8,19 @@ set(fortran_src
 
 if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
   set(CMAKE_Fortran_FLAGS
-      "-g -traceback -convert big_endian -free -assume byterecl -fp-model strict -mkl"
+      "-g -traceback -convert big_endian -free -assume byterecl -fp-model strict -mkl ${CMAKE_Fortran_FLAGS}"
   )
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
```
Comment (climbfuji) [inline src/CMakeLists.txt]: Just GNU

**Review by climbfuji — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -8,15 +8,19 @@ set(fortran_src
 
 if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
   set(CMAKE_Fortran_FLAGS
-      "-g -traceback -convert big_endian -free -assume byterecl -fp-model strict -mkl"
+      "-g -traceback -convert big_endian -free -assume byterecl -fp-model strict -mkl ${CMAKE_Fortran_FLAGS}"
```
Comment (aerorahul) [inline src/CMakeLists.txt]: I don't think we need `-mkl`.

File: src/CMakeLists.txt (hunk 1 above)
Comment (aerorahul) [inline src/CMakeLists.txt]: what @climbfuji said.

**Review by aerorahul — CHANGES_REQUESTED** [review]
GNU and -mkl comments

**Review by aerorahul — APPROVED** [review]

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -8,15 +8,19 @@ set(fortran_src
...
 if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
   set(CMAKE_Fortran_FLAGS
-      "-g -traceback -convert big_endian -free -assume byterecl -fp-model strict -mkl"
+      "-g -traceback -convert big_endian -free -assume byterecl -fp-model strict ${CMAKE_Fortran_FLAGS}"
   )
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
-elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
-  set(CMAKE_Fortran_FLAGS "-g -fconvert=big-endian -ffree-form")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "-g -fconvert=big-endian -ffree-form ${CMAKE_Fortran_FLAGS}")
```
Comment (DusanJovic-NOAA) [inline src/CMakeLists.txt]: I think `-ffree-form` is not needed.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -8,15 +8,19 @@ set(fortran_src
...
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
   set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall")
 endif()
 
+if(${CMAKE_Fortran_COMPILER_ID} MATCHES "^(GNU)$" AND ${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
+   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -w -fallow-argument-mismatch -fallow-invalid-boz")
```
Comment (DusanJovic-NOAA) [inline src/CMakeLists.txt]: Is '-fallow-invalid-boz' needed?

Comment (kgerheiser) [reply src/CMakeLists.txt]: No

File: src/CMakeLists.txt (hunk 3 above)
Comment (kgerheiser) [reply src/CMakeLists.txt]: You're right

**Pull request merged**